Sources/_StringProcessing/ByteCodeGen.swift

```diff
@@ -152,15 +154,16 @@ fileprivate extension Compiler.ByteCodeGen {
           }
         }
       } else {
-        builder.buildAssert { (input, pos, bounds) in
-          pos == input.startIndex
+        builder.buildAssert { (input, pos, subjectBounds) in
+          pos == subjectBounds.lowerBound
         }
       }
       
     case .endOfLine:
+      // FIXME: Anchor.endOfLine must always use this first branch
```

Comment:
What does this comment mean? Would it be relevant to someone trying to migrate this code over into the engine itself? How should they migrate it? Any XFAIL tests here?

Reply:
I'll rewrite the comment and add an XFAIL.